gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/SelectBtwModDataTimeBasedCopyableFileFilter.java

```diff
@@ -0,0 +1,116 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.gobblin.data.management.copy;
+
+import java.util.Collection;
+import java.util.Iterator;
+import java.util.Properties;
+
+import lombok.extern.slf4j.Slf4j;
+
+import org.apache.hadoop.fs.FileSystem;
+import org.joda.time.DateTime;
+import org.joda.time.DateTimeZone;
+import org.joda.time.Period;
+import org.joda.time.format.PeriodFormatter;
+import org.joda.time.format.PeriodFormatterBuilder;
+
+import com.google.common.collect.ImmutableList;
+
+import org.apache.gobblin.configuration.ConfigurationKeys;
+
+
+/**
+ * A {@link CopyableFileFilter} that drops a {@link CopyableFile} if file modified time not between the loop back window
+ *  <code>sourceFs<code>
+ */
+@Slf4j
+public class SelectBtwModDataTimeBasedCopyableFileFilter implements CopyableFileFilter {
+
+  private final Properties props;
+  private Period minLookBackPeriod;
+  private Period maxLookBackPeriod;
+  private DateTime currentTime;
+  private DateTime minLookBackTime;
+  private DateTime maxLookBackTime;
+
+  public static final String CONFIGURATION_KEY_PREFIX = "gobblin.dataset.";
+  public static final String MODIFIED_MIN_LOOK_BACK_TIME_KEY =
+      CONFIGURATION_KEY_PREFIX + "selection.modified.min.lookbackTime";
+  public static final String MODIFIED_MAX_LOOK_BACK_TIME_KEY =
+      CONFIGURATION_KEY_PREFIX + "selection.modified.max.lookbackTime";
+  public static final String DEFAULT_DATE_PATTERN_TIMEZONE = ConfigurationKeys.PST_TIMEZONE_NAME;
+  public static final String DATE_PATTERN_TIMEZONE_KEY = CONFIGURATION_KEY_PREFIX + "datetime.timezone";
+
+  public SelectBtwModDataTimeBasedCopyableFileFilter(Properties properties) {
+    this.props = properties;
+    PeriodFormatter periodFormatter =
+        new PeriodFormatterBuilder().appendDays().appendSuffix("d").appendHours().appendSuffix("h").toFormatter();
+    this.minLookBackPeriod = props.containsKey(MODIFIED_MIN_LOOK_BACK_TIME_KEY) ? periodFormatter.parsePeriod(
```

Comment:
i initially thought `minLookBackPeriod` as what `minLookBackPeriod` is.
If it helps, how about using startDate-endDate or since-by terminology? like
`this.modifiedSince = props.containsKey("gobblin.dataset.filter.modified.since")` or `this.modifiedStartDate = props.containsKey("gobblin.dataset.filter.modified.startDate")`

Reply:
i would like to follow the convention used in Gobblin to have min and max look back